alfresco-transformer-base/src/main/java/org/alfresco/transformer/QueueTransformService.java

```diff
@@ -94,75 +97,65 @@ public void receive(final Message msg)
 
         logger.info("New T-Request from queue with correlationId: {0}", correlationId);
 
+        Optional<TransformRequest> transformRequest;
+        try
+        {
+            transformRequest = convert(msg);
+        }
+        catch (TransformException e)
+        {
+            logger.error(e.getMessage(), e);
+            replyWithError(replyToDestinationQueue, HttpStatus.valueOf(e.getStatusCode()),
+                e.getMessage(), correlationId);
+            return;
+        }
 
-        TransformRequest transformRequest = convert(msg, correlationId, replyToDestinationQueue);
-        if (transformRequest == null)
+        if (!transformRequest.isPresent())
         {
-            logger.error("Exception during T-Request deserialization! T-Reply with error has been "
-                + "sent to T-Router!");
+            logger.error("T-Request is null deserialization!");
```

Comment:
Error messages should contain some detailed context (correlationId, requestId, sourceReference, ...) to allow log grep / tracing.